src/main/java/org/opensearch/ad/caching/CacheBuffer.java

```diff
@@ -240,16 +241,26 @@ public boolean canRemove() {
         ModelState<EntityModel> valueRemoved = items.remove(keyToRemove);
 
         if (valueRemoved != null) {
-            // if we releasing a shared cache item, release memory as well.
-            if (!reserved) {
-                memoryTracker.releaseMemory(memoryConsumptionPerEntity, false, Origin.MULTI_ENTITY_DETECTOR);
+            if (reserved) {
+                // release in reserved memory
+                memoryTracker.releaseMemory(memoryConsumptionPerEntity, true, Origin.HC_DETECTOR);
+            } else {
+                // release in shared memory
+                memoryTracker.releaseMemory(memoryConsumptionPerEntity, false, Origin.HC_DETECTOR);
             }
```

Comment:
Could be simplified to:
`memoryTracker.releaseMemory(memoryConsumptionPerEntity, reserved, Origin.HC_DETECTOR);`

Reply:
I realized I shouldn't release reserved memory.  Only keep the branch !reserved now.  Please check the new version.